packages/ember/barrel.ts

```diff
@@ -0,0 +1,751 @@
+import { getOnerror, setOnerror } from '@ember/-internals/error-handling';
+import EmberArray, {
+  A as EmberA,
+  NativeArray as EmberNativeArray,
+  isArray as emberIsArray,
+  makeArray as emberMakeArray,
+} from '@ember/array';
+import EmberMutableArray from '@ember/array/mutable';
+import EmberArrayProxy from '@ember/array/proxy';
+import EmberApplication, {
+  getOwner as applicationGetOwner,
+  setOwner as applicationSetOwner,
+  onLoad as applicationOnLoad,
+  runLoadHooks as applicationRunLoadHooks,
+} from '@ember/application';
+import EmberApplicationInstance from '@ember/application/instance';
+import EmberNamespace from '@ember/application/namespace';
+import EmberComponent, { Input as EmberInput } from '@ember/component';
+import EmberHelper from '@ember/component/helper';
+import EmberEngine from '@ember/engine';
+import EmberEngineInstance from '@ember/engine/instance';
+import EmberEnumerable from '@ember/enumerable';
+import EmberMutableEnumerable from '@ember/enumerable/mutable';
+import EmberCoreObject from '@ember/object/core';
+import EmberEvented, { on as emberOn } from '@ember/object/evented';
+import EmberMixin, { mixin as emberMixin } from '@ember/object/mixin';
+import EmberObservable from '@ember/object/observable';
+import {
+  addObserver as emberAddObserver,
+  removeObserver as emberRemoveObserver,
+} from '@ember/object/observers';
+import EmberObjectProxy from '@ember/object/proxy';
+import EmberPromiseProxyMixin from '@ember/object/promise-proxy-mixin';
+import EmberHashLocation from '@ember/routing/hash-location';
+import EmberHistoryLocation from '@ember/routing/history-location';
+import EmberNoneLocation from '@ember/routing/none-location';
+import EmberRoute from '@ember/routing/route';
+import EmberRouter from '@ember/routing/router';
+import {
+  controllerFor as emberControllerFor,
+  generateController as emberGenerateController,
+  generateControllerFactory as emberGenerateControllerFactory,
+  DSL as EmberRouterDSL,
+} from '@ember/routing/-internals';
+import {
+  isNone as emberIsNone,
+  isBlank as emberIsBlank,
+  isEmpty as emberIsEmpty,
+  isPresent as emberIsPresent,
+  isEqual as emberIsEqual,
+  typeOf as emberTypeOf,
+  compare as emberCompare,
+} from '@ember/utils';
+
+import * as glimmerRuntime from '@glimmer/runtime';
+
+import {
+  helperCapabilities,
+  setModifierManager,
+  setComponentTemplate,
+  getComponentTemplate,
+  setHelperManager,
+} from '@glimmer/manager';
+
+import {
+  assertDestroyablesDestroyed,
+  associateDestroyableChild,
+  destroy as emberDestroy,
+  enableDestroyableTracking,
+  isDestroying,
+  isDestroyed,
+  registerDestructor,
+  unregisterDestructor,
+} from '@ember/destroyable';
+
+import type { precompile, compile } from 'ember-template-compiler';
+import { _impl as EmberTestingImpl } from '@ember/test';
+import * as templateCompilation from '@ember/template-compilation';
+
+// eslint-disable-next-line @typescript-eslint/no-namespace
+namespace Ember {
```

Comment:
I have a hunch that some of these types may need to move back to the index.ts, but we'll see as the PR progresses

Reply:
Is this resolved?

Reply:
yup!